contracts/modules/cover/Cover.sol

```diff
@@ -0,0 +1,773 @@
+      // transfer the commission to the commissionDestination; reverts if commissionDestination is not a member
+      _tokenController.operatorTransfer(msg.sender, commissionDestination, commission);
+    }
+
+    _tokenController.burnFrom(msg.sender, price);
+  }
+
+  /* ========== Staking Pool creation ========== */
+
+  function createStakingPool(
+    address manager,
+    bool isPrivatePool,
+    uint initialPoolFee,
+    uint maxPoolFee,
+    ProductInitializationParams[] memory productInitializationParams,
+    uint depositAmount,
+    uint trancheId
+  ) external returns (address stakingPoolAddress) {
+
+    emit StakingPoolCreated(stakingPoolAddress, manager, stakingPoolImplementation);
+
+    CoverUtilsLib.PoolInitializationParams memory poolInitializationParams = CoverUtilsLib.PoolInitializationParams(
+      stakingPoolCount++,
+      manager,
+      isPrivatePool,
+      initialPoolFee,
+      maxPoolFee
+    );
+
+    return CoverUtilsLib.createStakingPool(
+      _products,
+      poolInitializationParams,
+      productInitializationParams,
+      depositAmount,
+      trancheId,
+      master.getLatestAddress("PS")
+    );
+  }
+
+  function stakingPool(uint index) public view returns (IStakingPool) {
+    return CoverUtilsLib.stakingPool(index, stakingPoolProxyCodeHash);
+  }
+
+  function coverData(uint coverId) external override view returns (CoverData memory) {
+    return _coverData[coverId];
+  }
+
+  function coverSegments(
+    uint coverId,
+    uint segmentId
+  ) public override view returns (CoverSegment memory) {
+    CoverSegment memory segment = _coverSegments[coverId][segmentId];
+    uint96 amountPaidOut = _coverData[coverId].amountPaidOut;
+    segment.amount = segment.amount >= amountPaidOut
+      ? segment.amount - amountPaidOut
+      : 0;
+    return segment;
+  }
+
+  function products(uint id) external override view returns (Product memory) {
+    return _products[id];
+  }
+
+  function productTypes(uint id) external override view returns (ProductType memory) {
+    return _productTypes[id];
+  }
+
+  function coverSegmentsCount(uint coverId) external override view returns (uint) {
+    return _coverSegments[coverId].length;
+  }
+
+  function productsCount() external override view returns (uint) {
+    return _products.length;
+  }
+
+  /*========== COVER EXPIRATION ========== */
+
+  function expireCovers(uint[] calldata coverIds) external {
+    for (uint i = 0; i < coverIds.length; i++) {
+      expireCover(coverIds[i]);
+    }
+  }
+
+  function expireCover(uint coverId) public {
+
+    require(coverAmountTrackingEnabled, "Cover expiring not enabled");
+
+
+    uint lastCoverSegmentIndex = _coverSegments[coverId].length - 1;
+    CoverSegment memory lastCoverSegment = coverSegments(coverId, lastCoverSegmentIndex);
+
+    require(!lastCoverSegment.expired, "Cover: Cover is already expired.");
+
+    if (lastCoverSegment.period + lastCoverSegment.start > block.timestamp) {
```

Comment:
Change this to a require statement so we'd know it's a no-op during the gas estimation, before actually sending a transaction that does nothing.

Reply:
Also the condition seems wrong, the period + start should be in the past, not in the future.

Reply:
done